core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/probabilistic/gaussian/Gaussian400EndpointModel.kt

```diff
@@ -20,94 +24,131 @@ import kotlin.random.Random
  * - Assumes *diagonal covariance*, i.e., each feature dimension is treated independently.
  *
  * ## Internals:
- * - Maintains two `Density` instances: one for class 200 and one for class 400.
+ * - Maintains two `Density` instances: one for class 400 and one for class not-400.
  * - Computes the log-likelihood of the input under each class-specific Gaussian.
  * - Combines likelihoods with class priors using Bayes' rule to calculate normalized posterior probabilities.
- * - Predicts based on the posterior probabilities (i.e., probabilities sum to 1).
+ * - Predicts based on the posterior probabilities.
  *
  * ## Assumptions:
  * - The input encoding (`InputEncoderUtils.encode`) produces consistent-length numeric vectors.
- * - The only two supported classes are 200 and 400.
- *
- * @param dimension the fixed dimensionality of the input feature vectors.
- * @param warmup the number of warmup updates to familiarize the classifier with at least a few true observations
- */
-class GaussianModel : AbstractAIModel() {
-
-    var density200: Density? = null
-    var density400: Density? = null
-
-    /** Must be called once to initialize the model properties */
-    fun setup(dimension: Int, warmup: Int) {
-        require(dimension > 0 ) { "Dimension must be positive." }
-        require(warmup > 0 ) { "Warmup must be positive." }
-        this.dimension = dimension
-        this.density200 = Density(dimension)
-        this.density400 = Density(dimension)
-        this.warmup = warmup
+  */
+class Gaussian400EndpointModel (
+    endpoint: Endpoint,
+    warmup: Int = 10,
+    dimension: Int? = null,
+    encoderType: EMConfig.EncoderType= EMConfig.EncoderType.NORMAL,
+    randomness: Randomness
+): AbstractProbabilistic400EndpointModel(endpoint, warmup, dimension, encoderType, randomness) {
+
+    private var density400: Density? = null
+    private var densityNot400: Density? = null
+
+    fun getDensity400(): Density? = density400
+    fun getDensityNot400(): Density? = densityNot400
```

Comment:
see previous comment regarding properties with private set but public get in Kotlin